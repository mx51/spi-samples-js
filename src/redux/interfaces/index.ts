import { IProductState } from '../reducers/ProductSlice/interfaces';
import { ITerminalState } from '../reducers/TerminalSlice/interfaces';

export interface ISamplePosState {
  terminals: ITerminalState;
  products: IProductState;
}
