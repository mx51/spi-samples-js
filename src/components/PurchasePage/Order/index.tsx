/* eslint-disable no-console */
import React, { useState } from 'react';
import { ListItemText, Box, Button, Divider, List, ListItem, Paper, Typography } from '@material-ui/core';

import { useDispatch, useSelector } from 'react-redux';
import { IProductSelector } from '../../../redux/reducers/ProductSlice/interfaces';
import { productsSelector, productSubTotalSelector } from '../../../redux/reducers/ProductSlice/productSelector';
import { clearAllProducts } from '../../../redux/reducers/ProductSlice/productSlice';
import currencyFormat from '../../../utils/common/intl/currencyFormatter';
import OrderLineItem from '../../OrderLineItem';
import OrderSubTotal from '../../OrderSubTotal';
import useStyles from './index.styles';

function Order(): React.ReactElement {
  const dispatch = useDispatch();
  const clearAllProductsAction = () => {
    dispatch(clearAllProducts());
  };
  const products: Array<IProductSelector> = useSelector(productsSelector);
  const subtotalAmount: number = useSelector(productSubTotalSelector);

  const [surchargeAmount, setSurchargeAmount] = useState(0);
  const [tipAmount, setTipAmount] = useState(0);
  const [cashoutAmount, setCashoutAmount] = useState(0);

  // Note: logs will be removed once key pad component is created REMINDER: to remove eslint in line 1

  console.log(setSurchargeAmount);
  console.log(setTipAmount);
  console.log(setCashoutAmount);

  const totalAmount = subtotalAmount + surchargeAmount + tipAmount + cashoutAmount;

  const classes = useStyles();
  return (
    <Box component={Paper} className={classes.root} display="flex" flexDirection="column">
      <Box display="flex" justifyContent="space-between" className={classes.heading}>
        <Typography variant="h6" component="h1">
          Order
        </Typography>
        <Box>
          <Button className={classes.clear} onClick={() => clearAllProductsAction()}>
            Clear all
          </Button>
        </Box>
      </Box>
      <Divider />
      <Box className={classes.orderList}>
        <List>
          {products.map((product) => (
            <ListItem key={product.product.id}>
              <ListItemText
                primary={`${product.quantity} x ${product.product.name}`}
                classes={{ primary: classes.items }}
              />
              <Typography className={classes.amount}>
                {currencyFormat((product.product.price * product.quantity) / 100)}
              </Typography>
            </ListItem>
          ))}
        </List>
      </Box>
      <Divider />
      <List>
        <OrderSubTotal label="Subtotal" amount={subtotalAmount} />
        <OrderLineItem label="Surcharge" amount={surchargeAmount} />
        <OrderLineItem label="Cashout" amount={cashoutAmount} />
        <OrderLineItem label="Tip" amount={tipAmount} />
        <Divider variant="middle" />
        <ListItem>
          <ListItemText primary="Total" classes={{ primary: classes.total }} />
          <Typography className={classes.totalPrice}>{currencyFormat(totalAmount / 100)}</Typography>
        </ListItem>
      </List>
      <Button
        variant="contained"
        color="primary"
        size="large"
        classes={{ root: classes.payNowBtn, label: classes.payNowBtnLabel }}
      >
        Pay now
      </Button>
    </Box>
  );
}

export default Order;
