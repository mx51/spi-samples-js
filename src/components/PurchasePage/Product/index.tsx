import React from 'react';
import { Box, Card, CardActionArea, CardContent, CardMedia, Typography } from '@material-ui/core';
import useStyles from './index.styles';
import { ProductProps } from './interfaces';

export default function Product({ product }: ProductProps): React.ReactElement {
  const classes = useStyles();
  return (
    <Box>
      <Card className={classes.cardStyle}>
        <CardActionArea>
          <CardMedia className={classes.productText} title={product.name}>
            <Box
              className={classes.productImage}
              bgcolor={product.backgroundColor}
              display="flex"
              justifyContent="center"
            >
              {product.image}
            </Box>
          </CardMedia>
          <CardContent>
            <Typography className={classes.productName} variant="h5" component="h2">
              {product.name}
            </Typography>
            <Typography variant="body2" color="textSecondary" component="p">
              {product.price}
            </Typography>
          </CardContent>
        </CardActionArea>
      </Card>
    </Box>
  );
}
