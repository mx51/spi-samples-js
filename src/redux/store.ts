import { AnyAction, Reducer, combineReducers, configureStore } from '@reduxjs/toolkit';
import { createMigrate, persistReducer, persistStore } from 'redux-persist';
import storage from 'redux-persist/lib/storage';
import hardSet from 'redux-persist/es/stateReconciler/hardSet';
import commonReducer from './reducers/CommonSlice/commonSlice';
import pairFormReducer from './reducers/PairFormSlice/pairFormSlice';
import { preAuthReducer } from './reducers/PreAuth/preAuthSlice';
import productSlice from './reducers/ProductSlice/productSlice';
import selectedTerminalReducer from './reducers/SelectedTerminalSlice/selectedTerminalSlice';
import terminalReducer from './reducers/TerminalSlice/terminalsSlice';
import { IPreAuthState } from './reducers/PreAuth/interfaces';
import { PayAtTableState, payAtTableReducer } from './reducers/PayAtTableSlice/payAtTableSlice';
import { ITerminalState } from './reducers/TerminalSlice/interfaces';

const terminalReducerMigration: Any = {
  0: (state: ITerminalState) => {
    Object.keys(state).forEach((key) => {
      if (state[key] && state[key].deviceAddress) {
        state[key].deviceAddress = state[key].deviceAddress.replace(/^wss?:\/\//, '');
      }
      if (state[key] && state[key].secrets) {
        state[key].secrets = {
          encKey: (state[key].secrets as Any).EncKey,
          hmacKey: (state[key].secrets as Any).HmacKey,
        };
      }
    });
    return state;
  },
};

const rootReducer = combineReducers({
  common: commonReducer,
  pairForm: pairFormReducer,
  products: productSlice,
  terminals: persistReducer<ITerminalState, AnyAction>(
    {
      key: 'terminals',
      version: 0,
      storage,
      stateReconciler: hardSet,
      migrate: createMigrate(terminalReducerMigration, { debug: false }),
    },
    terminalReducer
  ) as unknown as Reducer<ITerminalState>,
  selectedTerminal: selectedTerminalReducer,
  preAuth: persistReducer<IPreAuthState, AnyAction>(
    {
      key: 'root',
      storage,
      stateReconciler: hardSet,
    },
    preAuthReducer
  ) as unknown as Reducer<IPreAuthState>,
  payAtTable: persistReducer<PayAtTableState, AnyAction>(
    {
      key: 'payAtTable',
      storage,
      stateReconciler: hardSet,
    },
    payAtTableReducer
  ) as unknown as Reducer<PayAtTableState>,
});

export const store = configureStore({
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware({
      serializableCheck: false,
    }),
  reducer: rootReducer,
  devTools: process.env.NODE_ENV === 'development',
});

export const persistor = persistStore(store);

export type RootState = ReturnType<typeof rootReducer>;
export type AppStore = typeof store;
export type AppDispatch = typeof store.dispatch;
