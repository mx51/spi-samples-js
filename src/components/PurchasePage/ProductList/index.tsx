import React from 'react';
import { Box } from '@material-ui/core';
import products from '../../../definitions/constants/productList';
import Product from '../Product';

function ProductList(): React.ReactElement {
  return (
    <Box display="flex" flexDirection="coloum" flexWrap="wrap" justifyContent="space-around">
      {products.map((product) => (
        <Product key={product.name} product={product} />
      ))}
    </Box>
  );
}
export default ProductList;
