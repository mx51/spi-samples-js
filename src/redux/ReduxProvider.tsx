import React from 'react';
import { Provider } from 'react-redux';
// store
import { store } from './store';

interface ReduxProviderInterface {
  children: React.ReactNode;
  reduxStore: typeof store;
}

function ReduxProvider({ children, reduxStore }: ReduxProviderInterface): React.ReactElement {
  return <Provider store={reduxStore}>{children}</Provider>;
}

export default ReduxProvider;
