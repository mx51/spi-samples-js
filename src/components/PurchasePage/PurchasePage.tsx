import React from 'react';
import { Backdrop, Grid, Typography } from '@material-ui/core';
import { useSelector } from 'react-redux';
import { orderOverrideSubtotalSelector } from '../../redux/reducers/ProductSlice/productSelector';
import { isPaired } from '../../redux/reducers/TerminalSlice/terminalsSliceSelectors';
import Layout from '../Layout';
import NoTerminalPage from '../NoTerminalPage';
import useStyles from './index.styles';
import Order from './Order';
import ProductList from './ProductList';

function Purchase(): React.ReactElement {
  const classes = useStyles();
  const isTerminalPaired: boolean = useSelector(isPaired);
  const isOverrideSubtotalAmount = useSelector(orderOverrideSubtotalSelector);

  return (
    <Layout>
      <Grid container>
        <Grid item xs={8} className={classes.root}>
          <Backdrop className={classes.backdrop} open={isOverrideSubtotalAmount} />
          <Typography variant="h6" component="h1">
            Purchase
          </Typography>
          {!isTerminalPaired && <NoTerminalPage />}
          <ProductList />
        </Grid>
        <Grid item xs={4}>
          <Order disablePayNow={false} />
        </Grid>
      </Grid>
    </Layout>
  );
}

export default Purchase;
