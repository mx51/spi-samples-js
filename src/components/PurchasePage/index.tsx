import React from 'react';
import Button from '@material-ui/core/Button';
// component
import Layout from '../Layout';
// redux
import { incrementByAmount } from '../../redux/reducers/counterSlice';
import { useAppDispatch, useAppSelector } from '../../redux/hooks';

const Purchase: React.FC = () => {
  const count = useAppSelector((state) => state.counter.value);
  const dispatch = useAppDispatch();

  const handleClickCounter = () => {
    dispatch(incrementByAmount(5));
  };

  return (
    <Layout>
      <p>Purchase Page</p>
      <div>
        <Button onClick={handleClickCounter}>Counter is: {count}</Button>
      </div>
    </Layout>
  );
};

export default Purchase;
