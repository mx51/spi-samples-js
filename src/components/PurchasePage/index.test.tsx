import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';

import '@testing-library/jest-dom/extend-expect';
import PurchasePage from '.';
import ReduxProvider from '../../redux/ReduxProvider';
import { store } from '../../redux/store';

describe('Test <PurchasePage />', () => {
  test('snapshot for PurchasePage component', () => {
    const { container } = render(
      <ReduxProvider reduxStore={store}>
        <Router>
          <PurchasePage />
        </Router>
      </ReduxProvider>
    );
    expect(container).toMatchSnapshot();
  });
});
