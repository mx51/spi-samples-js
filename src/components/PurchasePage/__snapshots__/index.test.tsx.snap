// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <PurchasePage /> snapshot for PurchasePage component 1`] = `
<div>
  <div>
    <div
      class="makeStyles-root-2"
    >
      <header
        class="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary makeStyles-navbar-3 mui-fixed MuiPaper-elevation4"
      >
        <div
          class="MuiToolbar-root MuiToolbar-dense MuiToolbar-gutters"
        >
          <button
            aria-label="menu"
            class="MuiButtonBase-root MuiIconButton-root makeStyles-menuButton-5 MuiIconButton-colorInherit MuiIconButton-edgeStart"
            data-testid="navbarMenuIcon"
            name="menuIcon"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <svg>
            PrimaryLogoIcon.svg
          </svg>
        </div>
      </header>
    </div>
    <div
      class="MuiGrid-root MuiGrid-container"
    >
      <div
        class="MuiGrid-root makeStyles-root-1 MuiGrid-item MuiGrid-grid-xs-8"
      >
        <h1
          class="MuiTypography-root MuiTypography-h6"
        >
          Purchase
        </h1>
        <div
          class="MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-item"
          >
            <div
              class="MuiPaper-root makeStyles-paper-6 MuiPaper-elevation0 MuiPaper-rounded"
            >
              <svg>
                UnpluggedIcon.svg
              </svg>
              <h6
                class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-gutterBottom"
              >
                It looks like you don’t have any terminals paired yet
              </h6>
              <p
                class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
              >
                Fortunately, it’s very easy to pair a terminal and start using our Sample POS.
              </p>
              <a
                aria-disabled="false"
                class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
                href="/pair"
                role="button"
                tabindex="0"
              >
                <span
                  class="MuiButton-label"
                >
                  + Pair terminal
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </a>
            </div>
          </div>
        </div>
        <div
          class="MuiBox-root MuiBox-root-7"
        >
          <div
            class="MuiBox-root MuiBox-root-12"
          >
            <div
              class="MuiPaper-root MuiCard-root makeStyles-cardStyle-8 MuiPaper-elevation1 MuiPaper-rounded"
            >
              <button
                class="MuiButtonBase-root MuiCardActionArea-root"
                tabindex="0"
                type="button"
              >
                <div
                  class="MuiCardMedia-root makeStyles-productText-9"
                  title="Mocha"
                >
                  <div
                    class="MuiBox-root MuiBox-root-13 makeStyles-productImage-11"
                  >
                    <img
                      alt="Mocha"
                      src="images/product/mocha.png"
                    />
                  </div>
                </div>
                <div
                  class="MuiCardContent-root"
                >
                  <h2
                    class="MuiTypography-root makeStyles-productName-10 MuiTypography-h5"
                  >
                    Mocha
                  </h2>
                  <p
                    class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
                  >
                    $4.20
                  </p>
                </div>
                <span
                  class="MuiCardActionArea-focusHighlight"
                />
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
          <div
            class="MuiBox-root MuiBox-root-14"
          >
            <div
              class="MuiPaper-root MuiCard-root makeStyles-cardStyle-8 MuiPaper-elevation1 MuiPaper-rounded"
            >
              <button
                class="MuiButtonBase-root MuiCardActionArea-root"
                tabindex="0"
                type="button"
              >
                <div
                  class="MuiCardMedia-root makeStyles-productText-9"
                  title="Honey Latte"
                >
                  <div
                    class="MuiBox-root MuiBox-root-15 makeStyles-productImage-11"
                  >
                    <img
                      alt="Honey Latte"
                      src="images/product/honeyLatte.png"
                    />
                  </div>
                </div>
                <div
                  class="MuiCardContent-root"
                >
                  <h2
                    class="MuiTypography-root makeStyles-productName-10 MuiTypography-h5"
                  >
                    Honey Latte
                  </h2>
                  <p
                    class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
                  >
                    $4.80
                  </p>
                </div>
                <span
                  class="MuiCardActionArea-focusHighlight"
                />
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
          <div
            class="MuiBox-root MuiBox-root-16"
          >
            <div
              class="MuiPaper-root MuiCard-root makeStyles-cardStyle-8 MuiPaper-elevation1 MuiPaper-rounded"
            >
              <button
                class="MuiButtonBase-root MuiCardActionArea-root"
                tabindex="0"
                type="button"
              >
                <div
                  class="MuiCardMedia-root makeStyles-productText-9"
                  title="Flat White"
                >
                  <div
                    class="MuiBox-root MuiBox-root-17 makeStyles-productImage-11"
                  >
                    <img
                      alt="Flat White"
                      src="images/product/flatWhite.png"
                    />
                  </div>
                </div>
                <div
                  class="MuiCardContent-root"
                >
                  <h2
                    class="MuiTypography-root makeStyles-productName-10 MuiTypography-h5"
                  >
                    Flat White
                  </h2>
                  <p
                    class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
                  >
                    $3.80
                  </p>
                </div>
                <span
                  class="MuiCardActionArea-focusHighlight"
                />
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
          <div
            class="MuiBox-root MuiBox-root-18"
          >
            <div
              class="MuiPaper-root MuiCard-root makeStyles-cardStyle-8 MuiPaper-elevation1 MuiPaper-rounded"
            >
              <button
                class="MuiButtonBase-root MuiCardActionArea-root"
                tabindex="0"
                type="button"
              >
                <div
                  class="MuiCardMedia-root makeStyles-productText-9"
                  title="Chai Latte"
                >
                  <div
                    class="MuiBox-root MuiBox-root-19 makeStyles-productImage-11"
                  >
                    <img
                      alt="Chai Latte"
                      src="images/product/chaiLatte.png"
                    />
                  </div>
                </div>
                <div
                  class="MuiCardContent-root"
                >
                  <h2
                    class="MuiTypography-root makeStyles-productName-10 MuiTypography-h5"
                  >
                    Chai Latte
                  </h2>
                  <p
                    class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
                  >
                    $4.50
                  </p>
                </div>
                <span
                  class="MuiCardActionArea-focusHighlight"
                />
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
          <div
            class="MuiBox-root MuiBox-root-20"
          >
            <div
              class="MuiPaper-root MuiCard-root makeStyles-cardStyle-8 MuiPaper-elevation1 MuiPaper-rounded"
            >
              <button
                class="MuiButtonBase-root MuiCardActionArea-root"
                tabindex="0"
                type="button"
              >
                <div
                  class="MuiCardMedia-root makeStyles-productText-9"
                  title="Rose Tea Latte"
                >
                  <div
                    class="MuiBox-root MuiBox-root-21 makeStyles-productImage-11"
                  >
                    <img
                      alt="Rose Tea Latte"
                      src="images/product/roseTeaLatte.png"
                    />
                  </div>
                </div>
                <div
                  class="MuiCardContent-root"
                >
                  <h2
                    class="MuiTypography-root makeStyles-productName-10 MuiTypography-h5"
                  >
                    Rose Tea Latte
                  </h2>
                  <p
                    class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
                  >
                    $4.50
                  </p>
                </div>
                <span
                  class="MuiCardActionArea-focusHighlight"
                />
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
          <div
            class="MuiBox-root MuiBox-root-22"
          >
            <div
              class="MuiPaper-root MuiCard-root makeStyles-cardStyle-8 MuiPaper-elevation1 MuiPaper-rounded"
            >
              <button
                class="MuiButtonBase-root MuiCardActionArea-root"
                tabindex="0"
                type="button"
              >
                <div
                  class="MuiCardMedia-root makeStyles-productText-9"
                  title="Cappuccino"
                >
                  <div
                    class="MuiBox-root MuiBox-root-23 makeStyles-productImage-11"
                  >
                    <img
                      alt="Cappuccino"
                      src="images/product/cap.png"
                    />
                  </div>
                </div>
                <div
                  class="MuiCardContent-root"
                >
                  <h2
                    class="MuiTypography-root makeStyles-productName-10 MuiTypography-h5"
                  >
                    Cappuccino
                  </h2>
                  <p
                    class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
                  >
                    $3.80
                  </p>
                </div>
                <span
                  class="MuiCardActionArea-focusHighlight"
                />
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
          <div
            class="MuiBox-root MuiBox-root-24"
          >
            <div
              class="MuiPaper-root MuiCard-root makeStyles-cardStyle-8 MuiPaper-elevation1 MuiPaper-rounded"
            >
              <button
                class="MuiButtonBase-root MuiCardActionArea-root"
                tabindex="0"
                type="button"
              >
                <div
                  class="MuiCardMedia-root makeStyles-productText-9"
                  title="Latte"
                >
                  <div
                    class="MuiBox-root MuiBox-root-25 makeStyles-productImage-11"
                  >
                    <img
                      alt="Latte"
                      src="images/product/latte.png"
                    />
                  </div>
                </div>
                <div
                  class="MuiCardContent-root"
                >
                  <h2
                    class="MuiTypography-root makeStyles-productName-10 MuiTypography-h5"
                  >
                    Latte
                  </h2>
                  <p
                    class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
                  >
                    $3.80
                  </p>
                </div>
                <span
                  class="MuiCardActionArea-focusHighlight"
                />
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
          <div
            class="MuiBox-root MuiBox-root-26"
          >
            <div
              class="MuiPaper-root MuiCard-root makeStyles-cardStyle-8 MuiPaper-elevation1 MuiPaper-rounded"
            >
              <button
                class="MuiButtonBase-root MuiCardActionArea-root"
                tabindex="0"
                type="button"
              >
                <div
                  class="MuiCardMedia-root makeStyles-productText-9"
                  title="Production Test Item"
                >
                  <div
                    class="MuiBox-root MuiBox-root-27 makeStyles-productImage-11"
                  >
                    <img
                      alt="Production Test Item"
                      src="images/product/testItem.png"
                    />
                  </div>
                </div>
                <div
                  class="MuiCardContent-root"
                >
                  <h2
                    class="MuiTypography-root makeStyles-productName-10 MuiTypography-h5"
                  >
                    Production Test Item
                  </h2>
                  <p
                    class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
                  >
                    $0.01
                  </p>
                </div>
                <span
                  class="MuiCardActionArea-focusHighlight"
                />
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
      >
        <div
          class="MuiPaper-root MuiBox-root MuiBox-root-38 makeStyles-root-28 MuiPaper-elevation1 MuiPaper-rounded"
        >
          <div
            class="MuiBox-root MuiBox-root-39 makeStyles-heading-29"
          >
            <h1
              class="MuiTypography-root MuiTypography-h6"
            >
              Order
            </h1>
            <div
              class="MuiBox-root MuiBox-root-40"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-clear-30"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  Clear all
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
          <hr
            class="MuiDivider-root"
          />
          <div
            class="MuiBox-root MuiBox-root-41 makeStyles-orderList-37"
          >
            <ul
              class="MuiList-root MuiList-padding"
            />
          </div>
          <hr
            class="MuiDivider-root"
          />
          <ul
            class="MuiList-root MuiList-padding"
          >
            <li
              class="MuiListItem-root MuiListItem-gutters"
            >
              <div
                class="MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiListItemText-primary makeStyles-subTotal-42 MuiTypography-body1 MuiTypography-displayBlock"
                >
                  Subtotal
                </span>
              </div>
              <p
                class="MuiTypography-root makeStyles-price-43 MuiTypography-body1"
              >
                $0.00
              </p>
            </li>
            <li
              class="MuiListItem-root MuiListItem-gutters"
            >
              <div
                class="MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiListItemText-primary makeStyles-additionalCharges-44 MuiTypography-body1 MuiTypography-displayBlock"
                >
                  Surcharge
                </span>
              </div>
              <span
                aria-disabled="false"
                aria-label="Add Amount"
                class="MuiButtonBase-root MuiIconButton-root makeStyles-addAmountBtn-47"
                role="button"
                tabindex="0"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M13 7h-2v4H7v2h4v4h2v-4h4v-2h-4V7zm-1-5C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </span>
            </li>
            <li
              class="MuiListItem-root MuiListItem-gutters"
            >
              <div
                class="MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiListItemText-primary makeStyles-additionalCharges-44 MuiTypography-body1 MuiTypography-displayBlock"
                >
                  Cashout
                </span>
              </div>
              <span
                aria-disabled="false"
                aria-label="Add Amount"
                class="MuiButtonBase-root MuiIconButton-root makeStyles-addAmountBtn-47"
                role="button"
                tabindex="0"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M13 7h-2v4H7v2h4v4h2v-4h4v-2h-4V7zm-1-5C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </span>
            </li>
            <li
              class="MuiListItem-root MuiListItem-gutters"
            >
              <div
                class="MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiListItemText-primary makeStyles-additionalCharges-44 MuiTypography-body1 MuiTypography-displayBlock"
                >
                  Tip
                </span>
              </div>
              <span
                aria-disabled="false"
                aria-label="Add Amount"
                class="MuiButtonBase-root MuiIconButton-root makeStyles-addAmountBtn-47"
                role="button"
                tabindex="0"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M13 7h-2v4H7v2h4v4h2v-4h4v-2h-4V7zm-1-5C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </span>
            </li>
            <hr
              class="MuiDivider-root MuiDivider-middle"
            />
            <li
              class="MuiListItem-root MuiListItem-gutters"
            >
              <div
                class="MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiListItemText-primary makeStyles-total-31 MuiTypography-body1 MuiTypography-displayBlock"
                >
                  Total
                </span>
              </div>
              <p
                class="MuiTypography-root makeStyles-totalPrice-32 MuiTypography-body1"
              >
                $0.00
              </p>
            </li>
          </ul>
          <button
            class="MuiButtonBase-root MuiButton-root makeStyles-payNowBtn-33 MuiButton-contained MuiButton-containedPrimary MuiButton-containedSizeLarge MuiButton-sizeLarge"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label makeStyles-payNowBtnLabel-34"
            >
              Pay now
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
